### DISCUSSION TITLE
Static export of 404 page with app router

### DISCUSSION THREAD

**Discussion by wupr** [vercel/next.js#50345] [Help]
### Summary

I'm migrating from the pages router to the app router, and everything works except the 404 page.

I've moved my `/pages/404.js` to `/app/not-found.js`, but the 404.html exported is now the out-of-the-box Next.js 404 page, rather than my custom 404 page.

~~I'm not sure whether this is a bug or I'm missing something.~~
This seems to be a bug.

### Additional information

Output from `npx next info`:
```
    Operating System:
      Platform: linux
      Arch: x64
      Version: #1 SMP Fri Jan 27 02:56:13 UTC 2023
    Binaries:
      Node: 18.16.0
      npm: 9.6.6
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.4.3
      eslint-config-next: 13.4.3
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.0.4
```

### Example

For an MWE:
1. Use the default template generated by `create-next-app`
2. Add `not-found.js` to the app directory:
```js
export default function NotFound() {
    return (
        <main>
            <h1>404 Not Found</h1>
            <p>This is a custom 404 page.</p>
        </main>
    )
}
```
3. Modify `next.config.js`:
```js
/** @type {import('next').NextConfig} */
const nextConfig = {
    output: 'export',
}

module.exports = nextConfig
```

Now run `npm run build`. Partial output:
```
Route (app)                                Size     First Load JS
┌ ○ /                                      4.94 kB        82.1 kB
├ ○ /_not-found                            0 B                0 B
└ ○ /favicon.ico                           0 B                0 B
+ First Load JS shared by all              77.2 kB
  ├ chunks/139-d20cdafe45ea28b1.js         24.7 kB
  ├ chunks/2443530c-d4a6bd7fc7f21e3c.js    50.5 kB
  ├ chunks/main-app-0c88ea392f10c614.js    215 B
  └ chunks/webpack-fc75983a9531351a.js     1.68 kB

Route (pages)                              Size     First Load JS
─ ○ /404                                   178 B          83.9 kB
+ First Load JS shared by all              83.7 kB
  ├ chunks/main-1f6cce73b7f2f02b.js        81.8 kB
  ├ chunks/pages/_app-c544d6df833bfd4a.js  192 B
  └ chunks/webpack-fc75983a9531351a.js     1.68 kB

○  (Static)  automatically rendered as static HTML (uses no initial props)
```

Inspect `/out` to see that the only error page is `/out/404.html` and its content is not the custom message.

It may be worth noting that when running `npm run dev` and visiting localhost:3000/404 or a non-existent path, I see my custom 404 page.

**Comment by unleashit** [comment]
This is a bug that has been posted several times (example: https://github.com/vercel/next.js/issues/48227). They don't seem to have acknowledged it, so it doesn't seem like a priority at this point.

**Reply by wupr** [reply]
Ah, I guess I hadn't looked carefully enough in issues. Thanks for pointing me there!

**Comment by wupr** [comment]
Closing the discussion as this is now solved (see #48227).
